### DISCUSSION TITLE
WANTED: Frontend developers for Github FOSS project

### DISCUSSION THREAD

**Discussion by hwspeedy** [hardinfo2/hardinfo2#18] [General]
We are rebooting a nice Linux GUI tool as hardinfo2 - system information and benchmark for Linux.

That has a homepage that is just showing a proof-of-concept page - showing description of the program and some benchmarks that the users have sent in. (tables and graph.js)

https://hardinfo2.org/

It's a standard LAMP server that needs a frontend - eg. html, css, js coding - all basic

Open for all that want to learn or work together on FOSS development.

Server repo: https://github.com/hardinfo2/server

We are just releasing the app for all Linux distributions after more than 10 years with out releases and users are starting to pick up the homepage.

Please join the project, thanx.

**Comment by advitiya7** [comment]
Hello, i came here from reddit

**Comment by hwspeedy** [comment]
Hi @advitiya7,

Nice, that you want to help with homepage!

**Reply by advitiya7** [reply]
@hwspeedy can i start doing after a week, as my exams are scheduled in the next week sorry for the delay

**Comment by hwspeedy** [comment]
sure - education is more important.

I have invited you to the community

The page should could be seen on phone, tablets, PCs
Show info about hardinfo2 app
show links to github
maybe some examples of benchmarks like you see on hardinfo2.org - graph.js is at bottom.

**Reply by hwspeedy** [reply]
Also note @diazvictor has made something - I did not have setup ready for him and he got confused and mixed front/back-end. Stopped since...

https://github.com/hardinfo2/server/pull/2

**Reply by advitiya7** [reply]
For the frontend part can't we use react js with tailwind css it would be very helpful. @hwspeedy

**Reply by hwspeedy** [reply]
Frontend - any scripting framework, that you think most people can do - react.js sounds fine.

Backend is LAMP (PHP) - mostly we should try to just get all data once and then show different parts of them - it's not that many - see [api/getbenchmarks](https://hardinfo2.org/api/getbenchmarks)

web api examples in https://github.com/hardinfo2/server/tree/main/www/api

**Reply by hwspeedy** [reply]
After exams - please start development in /new2/ like described here https://github.com/hardinfo2/server/pull/2

Then we can test it.

Regarding found content - it needs to either be free and without mentioning or compatible with GPL2+.

You can ping me any time, but first - read more in the books for exams - best of luck!

**Reply by advitiya7** [reply]
@hwspeedy about backend I don't know php, but i can write code in react js the data can be fetched using react to no problem in that.

**Reply by hwspeedy** [reply]
Great - you just ping me and we will discuss and I will implement backend.

**Reply by hwspeedy** [reply]
here is a list of the organization starting to build up. https://github.com/hardinfo2/hardinfo2/discussions/16

**Reply by advitiya7** [reply]
hey @hwspeedy  i'm free now where do i start? 
and what all things are there that i need to know

**Comment by hwspeedy** [comment]
hi @advitiya7 
Hope all went well with the exams.

As we are building the initial webapp framework we should go in steps.

1. Get to know the program - it is a system information and benchmark program. Sends the benchmark data and some system information about the benchmark data to the server - They are picked up and put into the database. The webapp can then access these via web api.

2. I think we first need to get react framework up and running. All .js files must be local, no javascript in html, etc. - normal security practice.
we put the new source code it in ..www/new2/ to be able to see/feel it until the basic is ready.

3. Get a basic version running in the new framework - pretty much containing the same elements as the existing demo to show proof of concept with react.js.

4. UI/UX: Add the user interface with selections - benchmark type, graph/table, etc.

lots of fun - what ever you can think of that makes sense to the users.

Does that sound like a plan?

**Comment by advitiya7** [comment]
Do we have a figma design like what the frontend should look like? So that
i can start working on it, like we have to make a homepage so what all info
does it need to have

**Reply by hwspeedy** [reply]
**I am totally open as long as it is good for the 90% users of hardinfo2.**
Some ideas:
- start should maybe show overview of program with info and links to github - then some "hardinfo2 app" to go into details with the program.
- some buttons to go to benchmarks - there are many possibilities - overview, compare yours against world, against same type, compare some chosen, fastest of them all/record for categories, distributions compare speeds, etc. - The imagination is the limit.
- later we will add the history of hardinfo2 - how it was when linux was young and up to today - FOSS development/history preserve.

examples of other benchmark sites: 
https://www.userbenchmark.com/
https://www.cpubenchmark.net/
https://benchmarks.ul.com/3dmark
https://www.hwinfo.com/
https://www.phoronix-test-suite.com/ 

We are different as we are FOSS for linux.

**Reply by hwspeedy** [reply]
about with credits of the many people that helped make this possible.

**Reply by hwspeedy** [reply]
I also think that group benchmarks could be nice
 - all your machines
 - a community that wants to compare to each other, etc.


I am planning something like that in the app right now. 
It will be a work in progress and we do not know where we end up - just as long as it was fun getting there and it is good for the users.

**Comment by hwspeedy** [comment]
another later idea add PWA for phones - install web app as phone app - easy access - might give new ideas...

https://support.google.com/chrome/answer/9658361?hl=en&co=GENIE.Platform%3DAndroid

**Comment by hwspeedy** [comment]
debian has some requirements for info about the app: current version, changelog, news, etc. - I will be updated on what exactly they want and give info - It will probably all be something we pull from github release page (done in backend) (For frontend - just query a local web api and show it).

**Comment by advitiya7** [comment]
@hwspeedy  i'm really sorry, but if i am disturbing you please tell i still can't understand what all need to be done on the homepage like if i can get a picture or something that this is want we want in the homepage

**Comment by hwspeedy** [comment]
Hi @advitiya7 

Not at all - you just write and I will answer when I have time.

Are you talking about step 2 - proof of concept for basic framework with react.js?

No doubt that we can show pictures with a react.js framework.

Some idea of what I see the POF is about:
1. framework build up - how to work with react.js for a FOSS project.
2. Some nice idea for how to form the UI for a good UX on phone/pc/tablets with the knowledge we currently have (Large graphs/tables, lots of options, etc.).
3. The js graphs are not pictures - they are java scripts getting asynchronous data via fetch from web api with json and show dynamic interactive graphs.

**Reply by advitiya7** [reply]
i think you misunderstood me, like i will code with react so i need a idea that how the website will look  like a image or some design like this is what we want you to do

**Comment by hwspeedy** [comment]
The current page is just one long list of vanilla.js examples with full page (re)load - just a place holder for the real web app that will contain some of the elements and have multiple views and options, etc.

**Reply by advitiya7** [reply]
also please share the current website link

**Comment by hwspeedy** [comment]
The current webpage is mentioned at the top of this thread. 

https://hardinfo2.org

Coming from Server repo: https://github.com/hardinfo2/server/www

**Comment by hwspeedy** [comment]
**There are no boundary to how the webpage should look or will end up looking - just as long as it is good for the 90% users.**

maybe try something like  @diazvictor contribution - https://github.com/hardinfo2/server/pull/2 while loading
 - and then switch to program info like current (top of page)
    - extended with zooming pictures of the program if pushed. remove some pics if you like - you decide.
 - Then 3 pages with graphs from current page and 3 pages with tables from api/getbenchmarks
 - Some menubar with: welcome, hardinfo2 app, benchmarks->1,2,3,4,5,6, about->credits (Just put your name in center for now) 
 - just let it grow on you.

Maybe @diazvictor also wants to help with design - Do you know react.js? @diazvictor

**Reply by advitiya7** [reply]
i can make that design that @diazvictor  has uploaded no problem in that

**Reply by diazvictor** [reply]
>  i can make that design that @diazvictor  has uploaded no problem in that
> 

You can use it, no problem. I didn't follow it because of the limitations I started to notice and because of my responsibilities with the university.

Now that I am free and seeing that they are taking another approach (what I had in mind), I could give them a hand.

**Reply by diazvictor** [reply]
> Maybe @diazvictor also wants to help with design - Do you know react.js? @diazvictor

Of course. I will wait for the environment to be built and the project to be uploaded so I can collaborate

**Reply by hwspeedy** [reply]
@diazvictor - mega nice. thanx alot.

Just use this thread as communication at the beginning to be open to other who wants to contribute.
- Then you could probably all do pair programming and use discord and have fun - most important thing in FOSS and life 🥇

**Reply by advitiya7** [reply]
i will try to complete that basic layout, right now in india festivities are coming so we are busy with that sorry for any delay if any.
and i have to start the development in the server repo ? like the frontend part

**Reply by hwspeedy** [reply]
yes, both of you are frontend designers and will be making the framework for react.js for FOSS development. You can write about your strengths and coordinate right now so both of you can contribute - this is what FOSS does - it brings people together.

**Reply by hwspeedy** [reply]
We put the new react.js framework in server repo 
https://github.com/hardinfo2/server/tree/main/www/new1  - where @diazvictor gives input
https://github.com/hardinfo2/server/tree/main/www/new2 - where @advitiya7 gives input
You can both do contributions and then merge before putting it to www.

this is just temporary to get the basic react.js framework running - then it is moved  to www (Thereby being the new main page for hardinfo2).
